extensions/src/main/scala/water/parser/CategoricalPreviewParseWriter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package water.parser;
+
+import java.lang.reflect.Field;
+import water.util.IcedHashMap;
+
+public class CategoricalPreviewParseWriter extends PreviewParseWriter {
+
+  public CategoricalPreviewParseWriter(String[] domain, int totalCount) {
+    super(1);
+    this._nstrings[0] = totalCount;
+    IcedHashMap<String, String>[] domains = new IcedHashMap[1];
+    domains[0] = new IcedHashMapWrapper(domain);
+    setPrivateDomains(domains);
+  }
+
+  private void setPrivateDomains(IcedHashMap<String, String>[] domains) {
+    try {
```

Comment:
This should better not be done in parallel - non-atomic.

Reply:
Luckily, this is used by a single thread. https://github.com/h2oai/sparkling-water/pull/2341/files#diff-c8604610ecb616a35bcf8a8b81477e5fedd946767ca51d38316559057f34037bR95